062-support-tiered-storage.md

```diff
@@ -0,0 +1,153 @@
+```
+kafka:
+  tieredStorage:
+    type: custom
+    remoteStorageManager:
+      className: com.example.kafka.tiered.storage.s3.S3RemoteStorageManager
+      classPath: /opt/kafka/plugins/tiered-storage-s3/*
+      config:
+        # A map with String keys and String values. Key fields will be automatically prefixed with `rsm.config.` and appended to Kafka broker config.
+        storage.bucket.name: my-bucket
+  config:
+    ...
+    rlmm.config.remote.log.metadata.topic.replication.factor: 1
+```
+
+The configuration above will get translated to the below Kafka broker config:
+```
+# RLMM configuration generated by Strimzi
+remote.log.storage.system.enable: true
+remote.log.metadata.manager.impl.prefix: rlmm.config.
+remote.log.metadata.manager.class.name: org.apache.kafka.server.log.remote.metadata.storage.TopicBasedRemoteLogMetadataManager
+remote.log.metadata.manager.listener.name=REPLICATION-9091
+rlmm.config.remote.log.metadata.common.client.security.protocol=SSL
+rlmm.config.remote.log.metadata.common.client.ssl.keystore.location=/tmp/kafka/cluster.keystore.p12
+rlmm.config.remote.log.metadata.common.client.ssl.keystore.password=${CERTS_STORE_PASSWORD}
+rlmm.config.remote.log.metadata.common.client.ssl.keystore.type=PKCS12
+rlmm.config.remote.log.metadata.common.client.ssl.truststore.location=/tmp/kafka/cluster.truststore.p12
+rlmm.config.remote.log.metadata.common.client.ssl.keystore.password=${CERTS_STORE_PASSWORD}
+rlmm.config.remote.log.metadata.common.client.ssl.truststore.type=PKCS12
+#RSM configs set by the operator and by the user
+remote.log.storage.manager.class.name: com.example.kafka.tiered.storage.s3.S3RemoteStorageManager
+remote.log.storage.manager.class.path: /opt/kafka/plugins/tiered-storage-s3/*
+remote.log.storage.manager.impl.prefix: rsm.config.
+rsm.config.storage.bucket.name=my-bucket
+# Custom RLMM configs
+rlmm.config.remote.log.metadata.topic.replication.factor: 1
+```
+
+### Validation
+
+There are certain [limitations](https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Tiered+Storage+Early+Access+Release+Notes) for a tiered storage enabled cluster. At the time of this proposal writing, these are the known limitations:
+* Tiered storage is not supported on clusters utilizing the multiple Log directories on a broker (JBOD feature).
+* Currently, tiered storage is not available for compacted topics.
+* Once Tiered Storage is enabled for a topic, it becomes a permanent configuration and cannot be disabled.
+* Disabling Tiered Storage for the entire cluster requires manual deletion of all topics using Tiered Storage.
+* All Kafka clients, regardless of their version, can continue to produce and consume records from topics utilizing Tiered Storage. However, clients with versions earlier than 3.0 are limited in performing administrative actions, such as enabling Tiered Storage on a topic
+
+All the limitations above except the client compatibility are expected to be fixed before it is production-ready in Kafka.
```

Comment:
Sorry, I was wrong about this before. Some limitation might not get fixed before (or maybe after production-ready). I think it's enough to make it clear we won't do any additional thing for them in operator side. Could you please remove this line? Thanks.

Reply:
done